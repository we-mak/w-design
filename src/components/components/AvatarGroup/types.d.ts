import { GlobProps } from "../../../typings";

export interface AvatarGroupProps extends GlobProps {
  name?: string;
  description?: React.ReactChild | string;
  actions?: React.ReactChild | React.ReactChild[];
  avatar: React.ReactChild;
}
