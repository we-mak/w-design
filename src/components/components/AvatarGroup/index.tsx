import * as React from "react";
import styled from "styled-components";
import { AvatarProps } from "./types";
import Avartar from "../../elements/Avatar";

const AvatarGroupWrapper = styled.div`
  align-content: space-between;
  align-items: flex-start;
  display: flex;
`;
AvatarGroupWrapper.displayName = "AvatarGroupWrapper";

const AvatarWrapper = styled.div`
  flex: 0 0 auto;
`;
AvatarWrapper.displayName = "AvatarWrapper";

const AvatarContent = styled.div`
  flex: 1 1 auto;
  &:not(:last-child) {
    padding-right: 0.4rem;
  }
  &:not(:first-child) {
    padding-left: 0.4rem;
  }
`;
AvatarContent.displayName = "AvatarContent";

const AvatarTitle = styled.div`
  font-weight: 500;
  margin-bottom: 0.4rem;
`;
AvatarTitle.displayName = "AvatarTitle";

const AvatarDescription = styled.div`
  color: ${props => props.theme.colors["N40"]};
`;

const AvatarActions = styled.div`
  flex: 0 0 auto;
`;

const AvatarGroup = ({ name, description, actions, ...rest }: AvatarProps) => {
  return (
    <AvatarGroupWrapper>
      <AvatarWrapper>
        <Avartar dataInitial={name} {...rest} />
      </AvatarWrapper>
      <AvatarContent>
        {name && <AvatarTitle>{name}</AvatarTitle>}
        {description && <AvatarDescription>{description}</AvatarDescription>}
      </AvatarContent>
      {actions && <AvatarActions>{actions}</AvatarActions>}
    </AvatarGroupWrapper>
  );
};

export default AvatarGroup;
