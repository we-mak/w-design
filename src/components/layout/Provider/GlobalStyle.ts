import { createGlobalStyle } from "styled-components";
import theme from "../../../common/styleUtils/theme";
const colors = theme.colors;

/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
export const GlobalStyle = createGlobalStyle`
html {
  line-height: 1.15;
  -webkit-text-size-adjust: 100%;
}
body {
  margin: 0;
}
main {
  display: block;
}
h1 {
  font-size: 2em;
  margin: 0.67em 0;
}
hr {
  box-sizing: content-box;
  height: 0;
  overflow: visible;
}
pre {
  font-family: monospace, monospace;
  font-size: 1em;
}
a {
  background-color: transparent;
}
abbr[title] {
  border-bottom: none;
  text-decoration: underline;
  text-decoration: underline dotted;
}
b,
strong {
  font-weight: bolder;
}
code,
kbd,
samp {
  font-family: monospace, monospace;
  font-size: 1em;
}
small {
  font-size: 80%;
}
sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}
sub {
  bottom: -0.25em;
}
sup {
  top: -0.5em;
}
img {
  border-style: none;
}
button,
input,
optgroup,
select,
textarea {
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
}
button,
input {
  overflow: visible;
}
button,
select {
  text-transform: none;
}
[type="button"],
[type="reset"],
[type="submit"],
button {
  -webkit-appearance: button;
}
[type="button"]::-moz-focus-inner,
[type="reset"]::-moz-focus-inner,
[type="submit"]::-moz-focus-inner,
button::-moz-focus-inner {
  border-style: none;
  padding: 0;
}
[type="button"]:-moz-focusring,
[type="reset"]:-moz-focusring,
[type="submit"]:-moz-focusring,
button:-moz-focusring {
  outline: 1px dotted ButtonText;
}
fieldset {
  padding: 0.35em 0.75em 0.625em;
}
legend {
  box-sizing: border-box;
  color: inherit;
  display: table;
  max-width: 100%;
  padding: 0;
  white-space: normal;
}
progress {
  vertical-align: baseline;
}
textarea {
  overflow: auto;
}
[type="checkbox"],
[type="radio"] {
  box-sizing: border-box;
  padding: 0;
}
[type="number"]::-webkit-inner-spin-button,
[type="number"]::-webkit-outer-spin-button {
  height: auto;
}
[type="search"] {
  -webkit-appearance: textfield;
  outline-offset: -2px;
}
[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}
::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}
details {
  display: block;
}
summary {
  display: list-item;
}
template {
  display: none;
}
[hidden] {
  display: none;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
  font-size: 20px;
  line-height: 1.5;
  -webkit-tap-highlight-color: transparent;
}

body {
  font-size: .8rem;
  overflow-x: hidden;
  text-rendering: optimizeLegibility;
}


h1 {
    font-size: 1.6rem;
  }

  h2 {
    font-size: 1.4rem;
  }

  h3 {
    font-size: 1.2rem;
  }

  h4 {
    font-size: 1rem;
  }
  h5 {
    font-size: 0.9rem;
  }

  h6 {
    font-size: 0.8rem;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 700;
    line-height: 1.25;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.medium};
  }

  h1,
  h2,
  h3,
  h4 {
    color: ${colors.N50};
  }

  h5,
  h6 {
    color: ${colors.N60};
  }

  p {
    color: ${colors.N70};
    line-height: 1.5;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.normal};
  }

  a {
    font-family: ${(props: any) => props.theme.fonts.normal};
    color: ${colors.B50};
    outline: none;
    text-decoration: none;

    &:hover {
      color: ${colors.B30};
    }

    &:focus,
    &:hover,
    &:active {
      text-decoration: underline;
    }
    &:focus,
    &:active {
      color: ${colors.B60};
    }
    &:focus {
      text-decoration: none;
    }
  }

  b,
  strong {
    font-weight: bolder;
  }

  small {
    font-size: 80%;
    font-weight: 400;
    color: ${colors.N30};
  }

  blockquote {
    margin-left: 0;
    padding: 0.4rem 0.8rem;
    color: ${colors.N30};
    p:last-child {
      margin-bottom: 0;
    }
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
    text-align: left;
    width: 100%;
  }
  th {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N5"]};
    padding: 0.6rem 0.4rem;
    border-bottom-width: 0.1rem;
  }
  td {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N2"]};
    padding: 0.6rem 0.4rem;
  }
`;
