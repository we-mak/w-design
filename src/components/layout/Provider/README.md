Theme provider is based on [styled-components](https://www.styled-components.com/) and making theme with [styled-system](http://jxnblk.com/styled-system/)

Provider provides theme for your layout, you can supply your theme by insert your theme object by `theme` props
Default theme

```js static
defaultTheme = {
  breakpoint: ["24em", "32em", "48em", "64em", "80em"],
  fonts:
    '-apple-system, BlinkMacSystemFont, Helvetica, "Helvetica Neue", system-ui, sans-serif',
  fontSizes: [
    "0.7rem",
    "0.8rem",
    "0.9rem",
    "1rem",
    "1.2rem",
    "1.4rem",
    "1.6rem",
    "1.8rem",
    "2rem"
  ],
  fontWeights = { normal: 400, strong: 500, bold: 700, xbold: 900 },
  lineHeights = [1, 1.125, 1.25, 1.5],
  letterSpacings = { normal: "normal", caps: "0.25em" },
  radii = ["0rem", "0.1rem", "0.2rem", "0.3rem", "0.4rem"],
  space = [0, 4, 8, 16, 32, 64, 128], // space is used for margin and padding scales
  shadows = [
    "none",
    `0 .2rem .5rem ${hex2Rgba(colors.N50, 0.5)}`,
    `inset 0 0 0 1px rgba(87, 85, 217, .2), 0 0 4px rgba(87, 85, 217, .2)`
  ],
  colors: {...} // see colors
};
```

```js
<Provider>
  <h1>Default </h1>
  <p>
    Lorem ipsum
    <span>
      <a href="#">dolor </a>
    </span>
    sip amet
  </p>
</Provider>
```

```js
<Provider
  theme={{
    fonts: "Roboto, Helvetica, sans-serif"
  }}
>
  <h1>Roboto</h1>
  <p>
    Lorem ipsum
    <span>
      <a href="#">dolor </a>
    </span>
    sip amet
  </p>
</Provider>
```
