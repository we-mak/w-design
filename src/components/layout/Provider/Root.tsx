import styled from "../../../utils/styled";
import theme from "../../../common/theme";

const colors = theme.colors;

export const Root = styled.div`
  font-family: ${(props: any) => props.theme.fonts};

  h1 {
    font-size: 2rem;
  }

  h2 {
    font-size: 1.6rem;
  }

  h3 {
    font-size: 1.4rem;
  }

  h4 {
    font-size: 1.2rem;
  }
  h5 {
    font-size: 1rem;
  }

  h6 {
    font-size: 0.8rem;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 600;
    line-height: 1.25;
    margin-bottom: 0.5em;
    margin-top: 0;
  }

  h1,
  h2,
  h3,
  h4 {
    color: ${colors.N90};
  }

  h5,
  h6 {
    color: ${colors.N80};
  }

  p {
    color: ${colors.N70};
    line-height: 1.5;
    margin-bottom: 0.5em;
    margin-top: 0;
  }

  a {
    color: ${colors.B50};
    outline: none;
    text-decoration: none;

    &:hover {
      color: ${colors.B30};
    }

    &:focus,
    &:hover,
    &:active {
      text-decoration: underline;
    }
    &:focus,
    &:active {
      color: ${colors.B60};
    }
    &:focus {
      text-decoration: none;
    }
  }

  b,
  strong {
    font-weight: bolder;
  }

  code,
  kbd,
  pre,
  samp {
    font-family: "Menlo, Courier, monospace";
  }

  code {
    color: ${colors.R20};
    background: ${colors.N1};
    padding: 0.1rem 0.15rem;
    line-height: 1.2;
  }

  small {
    font-size: 80%;
    font-weight: 400;
    color: ${colors.N30};
  }

  blockquote {
    margin-left: 0;
    padding: 0.4rem 0.8rem;
    color: ${colors.N30};
    p:last-child {
      margin-bottom: 0;
    }
  }
`;

Root.displayName = "Root";

export default Root;
