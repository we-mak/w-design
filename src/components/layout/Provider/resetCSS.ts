import { colors } from "../../../common/colors";

const ResetCss = `
*,
*::before,
*::after {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
  font-size: 20px;
  line-height: 1.5;
  -webkit-tap-highlight-color: transparent;
}

body {
  background: ${colors.WHITE};
  color: ${colors.N60};
  font-family: -apple-system, BlinkMacSystemFont, Helvetica, "Helvetica Neue", system-ui, sans-serif;
  font-size: 0.8rem;
  overflow-x: hidden;
  text-rendering: optimizeLegibility;
}

a {
  font-family: Helvetica, "Helvetica Neue", system-ui, sans-serif;
}
`;

export default ResetCss;
