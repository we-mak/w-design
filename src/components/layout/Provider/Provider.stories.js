import React from "react";
import { storiesOf } from "@storybook/react";
import Provider from "./Provider";

storiesOf("components/Provider", module).addWithJSX("default Provider", () => (
  <Provider>
    <h1>Default </h1>
    <p>
      Lorem ipsum
      <span>
        <a href="#"> dolor </a>
      </span>{" "}
      sip amet
    </p>
  </Provider>
));
