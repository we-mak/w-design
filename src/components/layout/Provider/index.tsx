import * as React from "react";
import { ThemeProvider } from "styled-components";
import defaultTheme from "../../../common/theme";
import { ResetCss } from "./ResetCss";
import Root from "./Root";

export interface Props extends React.HTMLProps<HTMLDivElement> {
  theme?: any;
}

class Provider extends React.Component<Props> {
  stylesheet: any;

  static displayName = "WProvider";

  public static defaultProps: Props = {
    theme: {}
  };

  componentDidMount() {
    this.stylesheet = document.createElement("style");
    this.stylesheet.type = "text/css";
    // this.stylesheet.setAttribute("rel", "preload");
    // this.stylesheet.setAttribute("as", "style");
    this.stylesheet.innerHTML = ResetCss;

    if (document && document.head) {
      document.head.appendChild(this.stylesheet);
    }
  }

  componentWillUnmount() {
    if (this.stylesheet && document && document.head) {
      document.head.removeChild(this.stylesheet);
      delete this.stylesheet;
    }
  }

  render() {
    const { theme, ...props } = this.props;
    return (
      <ThemeProvider theme={{ ...defaultTheme, ...theme }}>
        <Root {...props as any} />
      </ThemeProvider>
    );
  }
}

export default Provider;
