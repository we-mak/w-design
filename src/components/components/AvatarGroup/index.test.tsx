import * as React from "react";
import { render } from "react-testing-library";
import "jest-styled-components";
import AvatarGroup from "./index";
import Provider from "../../layout/Provider";

describe("<AvatarGroup/>", () => {
  // Render test
  it("should renders correctly", () => {
    const container = render(
      <Provider>
        <AvatarGroup />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
