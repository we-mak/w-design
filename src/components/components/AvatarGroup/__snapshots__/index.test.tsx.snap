// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarGroup/> should renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  border-radius: 50%;
  color: rgba(255,255,255,0.85);
  display: inline-block;
  line-height: 1.25;
  margin: 0;
  position: relative;
  vertical-align: middle;
  height: 2.8rem;
  width: 2.8rem;
}

.c4 {
  color: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 1;
  text-transform: uppercase;
  font-weight: 500;
  font-size: 1rem;
}

.c1 {
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c5:not(:last-child) {
  padding-right: 0.4rem;
}

.c5:not(:first-child) {
  padding-left: 0.4rem;
}

.c6 {
  font-weight: 500;
  margin-bottom: 0.2em;
}

.c7 {
  color: #4D6171;
  font-size: 0.7rem;
}

.c8 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 h1 {
  font-size: 1.6rem;
}

.c0 h2 {
  font-size: 1.4rem;
}

.c0 h3 {
  font-size: 1.2rem;
}

.c0 h4 {
  font-size: 1rem;
}

.c0 h5 {
  font-size: 0.9rem;
}

.c0 h6 {
  font-size: 0.8rem;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  font-weight: 700;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-family: Helvetica,Arial,sans-serif;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4 {
  color: #394B59;
}

.c0 h5,
.c0 h6 {
  color: #293742;
}

.c0 p {
  color: #202B33;
  line-height: 1.5;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-family: Helvetica,Arial,sans-serif;
}

.c0 a {
  font-family: Helvetica,Arial,sans-serif;
  color: #008CC0;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  color: #00A7E4;
}

.c0 a:focus,
.c0 a:hover,
.c0 a:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 a:focus,
.c0 a:active {
  color: #007099;
}

.c0 a:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 b,
.c0 strong {
  font-weight: bolder;
}

.c0 small {
  font-size: 80%;
  font-weight: 400;
  color: #566E82;
}

.c0 blockquote {
  margin-left: 0;
  padding: 0.4rem 0.8rem;
  color: #566E82;
}

.c0 blockquote p:last-child {
  margin-bottom: 0;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
  text-align: left;
  width: 100%;
}

.c0 th {
  border-bottom: 0.05rem solid #BECEDC;
  padding: 0.6rem 0.4rem;
  border-bottom-width: 0.1rem;
}

.c0 td {
  border-bottom: 0.05rem solid #EFF3F6;
  padding: 0.6rem 0.4rem;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <figure
              class="c3"
              style="background-color: rgb(157, 179, 198);"
            >
              <span
                class="c4"
              >
                wd
              </span>
            </figure>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6"
            >
              we-design
            </div>
            <div
              class="c7"
            >
              this is the description
            </div>
          </div>
          <div
            class="c8"
          >
            <div>
              do action
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzXfLP cukYDc"
    >
      <div
        class="sc-AykKG kyjCvr"
      >
        <div
          class="sc-AykKH fsIIiH"
        >
          <figure
            class="sc-AykKC hTAlyy"
            style="background-color: rgb(157, 179, 198);"
          >
            <span
              class="sc-AykKD gkSYHo"
            >
              wd
            </span>
          </figure>
        </div>
        <div
          class="sc-AykKI eLQCqT"
        >
          <div
            class="sc-AykKJ jeYZkL"
          >
            we-design
          </div>
          <div
            class="sc-AykKK fHUmLc"
          >
            this is the description
          </div>
        </div>
        <div
          class="sc-fzXfLO euaKtg"
        >
          <div>
            do action
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
